src/Javis.jl

```diff
@@ -306,39 +343,220 @@ function get_javis_frame(video, objects, frame)
+        )
+            continue
+        end
+
+        # check if the object should be part of the global layer (i.e Background)
+        # or in its own layer (default)
+        in_global_layer = get(object.opts, :in_global_layer, false)::Bool
+        in_local_layer = get(object.opts, :in_local_layer, false)::Bool
+        if !in_global_layer && !in_local_layer
+            @layer begin
+                draw_object(object, video, frame, origin_matrix, layer_frames)
             end
+        else
+            draw_object(object, video, frame, origin_matrix, layer_frames)
+            # update origin_matrix as it's inside the global layer
+            origin_matrix = cairotojuliamatrix(getmatrix())
         end
+
         # if object is in global layer this changes the background settings
         update_background_settings!(background_settings, object)
     end
+end
+
+"""
+    get_layer_frame(video, layer, frame)
+
+Is called inside [`get_javis_frame`](@ref) and does two things viz.
+    - Creates a Luxor Drawing and renders the object of each layer
+    - computes the actions applies on the layer and stores them
+
+Returns the Drawing of the layer as an image matrix.
+"""
+function get_layer_frame(video, layer, frame)
+    Drawing(layer.width, layer.height, :image)
+    layer_frames = layer.frames
+    render_objects(layer.layer_objects, video, frame, layer_frames = layer_frames)
+
+    if frame in get_frames(layer)
+        # call currently active actions and their transformations for each layer
+        actions = layer.actions
+        for action in actions
+            get_frames(action) isa Nothing &&
+                error("Frame range for the layer's action might be missing")
+            rel_frame = frame - first(layer_frames.frames) + 1
+            if rel_frame in get_frames(action)
+                action.func(video, layer, action, rel_frame)
+            elseif rel_frame > last(get_frames(action)) && action.keep
+                # call the action on the last frame i.e. disappeared things stay disappeared
+                action.func(video, layer, action, last(get_frames(action)))
+            end
+        end
+    end
+    img_layer = image_as_matrix()
+    finish()
+    return img_layer
+end
+
+
+"""
+    apply_layer_settings(layer, pt)
+
+Applies the computed actions to the image matrix of the layer to it's image matrix
+Actions supported:
+- `anim_translate`:translates the entire layer to a specified position
+- `setopacity`:changes the opacity of the entire layer
+- `anim_rotate`:rotates the layer by a given angle 
+- `appear(:fade)`:fades in the layer
+- `disappear(:fade)`:fades out the layer
+
+It reads and applies the layer settings(computed by [`get_layer_frame`](@ref) function))
+"""
+function apply_layer_settings(layer_settings)
+    # final actions on the layer are applied here
+    # currently scale and translate are support
+    scale(layer_settings.scale)
+    rotate(layer_settings.rotation_angle)
```

Comment:
Changing it to 
```julia
translate(layer.position)
rotate(layer_settings.rotation_angle)
translate(-layer.position)
```
should fix the issue we discussed yesterday.

Reply:
>translate(-layer.position)

Damn. IK that it's just a `-` but how does it work?!

Reply:
As far as I understand the code:
You create an image matrix for the layer then create an extra drawing for all layers. In that general layer you apply the settings of each layer and then use `placeimage` to put it into the right spot? 
The problem is that you rotate the layer inside the general layer beforehand which changes the position of the global coordinate system and then you place the image there. What you want to do is shift the origin of the layer to the center first then rotate around that center. Afterwards shift it back.
You probably want to put the `translate(layer.position)` before the `scale` as it scales from the origin as well but I haven't tested yet.